sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -1376,6 +1376,28 @@ def test_encoders_string_categories(input_dtype, category_dtype, array_type):
     assert_array_equal(X_trans, expected)
 
 
+def test_mixed_string_bytes_categoricals():
+    """Check that this mixture of predefined categories and X raises an error.
+
+    Categories defined as bytes can not easily be compared to data that is
+    a string.
+    """
+    # data as unicode
+    X = np.array([["b"], ["a"]], dtype="U")
+    # predefined categories as bytes
+    categories = [np.array(["b", "a"], dtype="S")]
+    ohe = OneHotEncoder(categories=categories, sparse_output=False)
+
+    with pytest.raises(
+        ValueError,
+        match=(
+            "Found incompatible types ('bytes' vs 'str_') between values and predefined"
```

Comment:
Could you declare this string just before the context manager. It will be easier to read.